tests/integration/suite/actors/reminders/scheduler/metrics.go

```diff
@@ -0,0 +1,210 @@
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	clientv3 "go.etcd.io/etcd/client/v3"
+
+	runtimev1pb "github.com/dapr/dapr/pkg/proto/runtime/v1"
+	"github.com/dapr/dapr/tests/integration/framework"
+	clients "github.com/dapr/dapr/tests/integration/framework/client"
+	frameworkclient "github.com/dapr/dapr/tests/integration/framework/client"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	"github.com/dapr/dapr/tests/integration/framework/process/http/app"
+	"github.com/dapr/dapr/tests/integration/framework/process/placement"
+	"github.com/dapr/dapr/tests/integration/framework/process/ports"
+	"github.com/dapr/dapr/tests/integration/framework/process/scheduler"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(metrics))
+}
+
+type metrics struct {
+	place     *placement.Placement
+	scheduler *scheduler.Scheduler
+	triggered atomic.Int64
+
+	daprd    *daprd.Daprd
+	etcdPort int
+}
+
+func (m *metrics) Setup(t *testing.T) []framework.Option {
+	configFile := filepath.Join(t.TempDir(), "config.yaml")
+	require.NoError(t, os.WriteFile(configFile, []byte(`
+apiVersion: dapr.io/v1alpha1
+kind: Configuration
+metadata:
+  name: schedulerreminders
+spec:
+  features:
+  - name: SchedulerReminders
+    enabled: true`), 0o600))
+
+	fp := ports.Reserve(t, 2)
+	port1 := fp.Port(t)
+	port2 := fp.Port(t)
+	m.etcdPort = port2
+	clientPorts := []string{
+		"scheduler-0=" + strconv.Itoa(m.etcdPort),
+	}
+	m.scheduler = scheduler.New(t,
+		scheduler.WithID("scheduler-0"),
+		scheduler.WithInitialCluster(fmt.Sprintf("scheduler-0=http://localhost:%d", port1)),
+		scheduler.WithInitialClusterPorts(port1),
+		scheduler.WithEtcdClientPorts(clientPorts),
+	)
+
+	app := app.New(t,
+		app.WithHandlerFunc("/actors/myactortype/myactorid/method/remind/remindermethod", func(http.ResponseWriter, *http.Request) {
+			m.triggered.Add(1)
+		}),
+		app.WithHandlerFunc("/actors/myactortype/myactorid/method/foo", func(http.ResponseWriter, *http.Request) {}),
+		app.WithConfig(`{"entities": ["myactortype"]}`),
+	)
+
+	m.place = placement.New(t)
+
+	m.daprd = daprd.New(t,
+		daprd.WithConfigs(configFile),
+		daprd.WithInMemoryActorStateStore("mystore"),
+		daprd.WithPlacementAddresses(m.place.Address()),
+		daprd.WithSchedulerAddresses(m.scheduler.Address()),
+		daprd.WithAppPort(app.Port()),
+	)
+
+	fp.Free(t)
+	return []framework.Option{
+		framework.WithProcesses(app, m.scheduler, m.place, m.daprd),
+	}
+}
+
+func (m *metrics) Run(t *testing.T, ctx context.Context) {
+	m.scheduler.WaitUntilRunning(t, ctx)
+	m.place.WaitUntilRunning(t, ctx)
+	m.daprd.WaitUntilRunning(t, ctx)
+
+	frameworkClient := frameworkclient.HTTP(t)
+
+	client := m.daprd.GRPCClient(t, ctx)
+
+	etcdClient := clients.Etcd(t, clientv3.Config{
+		Endpoints:   []string{fmt.Sprintf("localhost:%d", m.etcdPort)},
+		DialTimeout: 5 * time.Second,
+	})
```

Comment:
```suggestion
	etcdClient :=  m.scheduler.ETCDClient(t, ctx)
```

Add the following to `/framework/process/scheduler/scheduler.go`

```go
func (s *Scheduler) ETCDClient(t *testing.T) *clientv3.Client {
       t.Helper()

       client, err := clientv3.New(clientv3.Config{
               Endpoints:   []string{fmt.Sprintf("127.0.0.1:%s", s.EtcdClientPort())},
               DialTimeout: 40 * time.Second,
       })
       require.NoError(t, err)

       t.Cleanup(func() {
               require.NoError(t, client.Close())
       })

       return client
}
```